test/spec.wizard-model.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const WizardModel = require('../lib/wizard-model');
+const JourneyModel = require('../lib/journey-model');
+const SessionModel = require('../lib/model');
+
+describe('session model', () => {
+    let wizardModel, journeyModel, session, fields;
+
+    beforeEach(() => {
+        sinon.stub(JourneyModel.prototype, 'registerModel');
+        sinon.stub(JourneyModel.prototype, 'set');
+        sinon.stub(JourneyModel.prototype, 'get').withArgs('foo').returns('bar');
+        session = {};
+        journeyModel = new JourneyModel(null, { key: 'journey', session });
+        fields = {
+            a: {},
+            b: { journeyKey: 'foo' },
+            c: { default: 'baz'},
+            d: { journeyKey: 'boo', default: 'bam' }
+        };
+
+        wizardModel = new WizardModel(null, {
+            key: 'test',
+            session,
+            journeyModel,
+            fields
+        });
+    });
+
+    afterEach(() => {
+        JourneyModel.prototype.registerModel.restore();
+        JourneyModel.prototype.get.restore();
+        JourneyModel.prototype.set.restore();
+    });
+
+    it('exports a function', () => {
+        WizardModel.should.be.a.function;
+    });
+
+    it('should be an instance of SessionModel', () => {
+        wizardModel.should.be.an.instanceOf(SessionModel);
+    });
+
+    describe('constructor', () => {
+        it('should throw an error if no options are provided', () => {
+            expect(() => new WizardModel()).to.throw();
+        });
+
+        it('should throw an error if no journeyModel is provided in options', () => {
```

Comment:
Could we also get a test that checks to see that journeyModel is an instance of SessionModel?